app/models/concerns/alegre_v2.rb

```diff
@@ -263,26 +327,133 @@ def get_confirmed_items(project_media, field)
       get_items(project_media, field, true)
     end
 
+    def get_suggested_items_async(project_media, field)
+      get_items_async(project_media, field)
+    end
+
+    def get_confirmed_items_async(project_media, field)
+      get_items_async(project_media, field, true)
+    end
+
     def get_similar_items_v2(project_media, field)
       type = get_type(project_media)
-      if !Bot::Alegre.should_get_similar_items_of_type?('master', project_media.team_id) || !Bot::Alegre.should_get_similar_items_of_type?(type, project_media.team_id)
+      if !should_get_similar_items_of_type?('master', project_media.team_id) || !should_get_similar_items_of_type?(type, project_media.team_id)
         {}
       else
         suggested_or_confirmed = get_suggested_items(project_media, field)
         confirmed = get_confirmed_items(project_media, field)
-        Bot::Alegre.merge_suggested_and_confirmed(suggested_or_confirmed, confirmed, project_media)
+        merge_suggested_and_confirmed(suggested_or_confirmed, confirmed, project_media)
+      end
+    end
+
+    def get_similar_items_v2_async(project_media, field)
+      type = get_type(project_media)
+      if !should_get_similar_items_of_type?('master', project_media.team_id) || !should_get_similar_items_of_type?(type, project_media.team_id)
+        return false
+      else
+        get_suggested_items_async(project_media, field)
+        get_confirmed_items_async(project_media, field)
+        return true
+      end
+    end
+
+    def get_required_keys(project_media, field)
+      {
+        confirmed_results: "alegre:async_results:#{project_media.id}_#{field}_true",
+        suggested_or_confirmed_results: "alegre:async_results:#{project_media.id}_#{field}_false"
+      }
+    end
+
+    def get_cached_data(required_keys)
+      redis = Redis.new(REDIS_CONFIG)
+      Hash[required_keys.collect{|k,v| [k, (Hash[YAML.load(redis.get(v)).collect{|kk,vv| [kk.to_i, vv]}] rescue [])]}]
```

Comment:
Can you please just add a comment explaining what this line is doing? A lot of Ruby magic here hehe.